### DISCUSSION TITLE
V5 Roadmap

### DISCUSSION THREAD

**Discussion by andrii-bodnar** [lingui/js-lingui#1872] [Ideas]
It's been a while since the latest v4 release and I'm suggesting to start gathering a list of things that might be included in the future v5 major release:

- [x] https://github.com/lingui/js-lingui/issues/1361
- [x] https://github.com/lingui/js-lingui/issues/1698
- [x] <s>Stable Dependency tree crawling</s>
- [x] Consider a better whitespace processing algorithm (https://github.com/lingui/js-lingui/discussions/1873)
- [x] Remove the `isTranslated` property from `TransNoContext` (#1907)
- [x] Remove other deprecations
- [x] [Node 20](https://endoflife.date/nodejs)
- [x] Update examples
- [x] Review the documentation #2078
  - [x] Introduction #2062
  - [x] Unify Installation instructions #2060
  - [x] Tutorials https://github.com/lingui/js-lingui/pull/2075
  - [x] Guides #2063
  - [x] API Reference #2055 #2065
  - [x] Tooling #2065
  - [x] Sync & Collaboration Tools #2061
  - [x] Resources #2064
  - [x] v5 Migration Guide https://github.com/lingui/js-lingui/pull/2077
- [x] Blog Post #2079

**Comment by EnablingLogic** [comment]
I don't know if this is the place to ask this, but would it be possible to consider taking the SWC plugin out of beta?

**Reply by timofei-iatsenko** [reply]
From our side we consider it as stable, but from the SWC itself plugins are still experimental, so we could do nothing about it.

**Reply by EnablingLogic** [reply]
Ah, that's good to know! Lingui seems to be the only library to take SWC seriously, so it'd be a shame if support for `@lingui/swc-plugin` would be dropped.

**Reply by timofei-iatsenko** [reply]
I'm going to support it as long as I can. No plans to drop it.

**Comment by trevcodesjs** [comment]
Are there any plans to move the experimental extractor out of the experimental phase?

**Reply by timofei-iatsenko** [reply]
We've discussed it today, and we are not plan to do it for V5 version. Because to make it properly some refactoring should be done to the lingui packages. We also didn't gather enough feedback to confidently say that this is not experimental anymore. 

Pls give it a try, and let us know how it worked for you. We need that feedback from community to make it stable.

**Comment by andrii-bodnar** [comment]
Have some ideas for improving the documentation. At the same time going to review everything before the v5 release.

**Reply by andrii-bodnar** [reply]
Also, I plan to write a v5 Blog post.

**Comment by PatrickRogg** [comment]
Hey guys,

thanks for your work on the v5 release. Would it be possible to make the v5 of [lingui/swc-plugin](https://github.com/lingui/swc-plugin/tree/next) compatible with swc core `v0.98.x`?

I can use the v5 of the swc-plugin with next.js `15.0.0-canary.195`, but in my node.js backend I'm told that the swc_core version `1.7.36` is not compatible with `"@lingui/swc-plugin": "^5.0.0-next.1",`

```
Caused by:
0: failed to invoke `@lingui/swc-plugin` as js transform plugin at @lingui/swc-plugin
1: failed to run Wasm plugin transform. Please ensure the version of `swc_core` used by the plugin is compatible with the host runtime. See the documentation for compatibility information. If you are an author of the plugin, please update `swc_core` to the compatible version.
```

Thanks for your help.

**Reply by andrii-bodnar** [reply]
Hello, It's probably related to this - https://github.com/lingui/swc-plugin/issues/119
